include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -62,20 +62,18 @@ namespace LinearAlgebra
     {
       if (new_alloc_size > allocated_size)
         {
-          Assert (((allocated_size > 0 && val != nullptr) ||
-                   val == nullptr), ExcInternalError());
-          if (val != nullptr)
-            free(val);
+          Assert (((allocated_size > 0 && values != nullptr) ||
+                   values == nullptr), ExcInternalError());
 
-          Utilities::System::posix_memalign ((void **)&val, 64, sizeof(Number)*new_alloc_size);
+          Number *new_val;
+          Utilities::System::posix_memalign ((void **)&new_val, 64, sizeof(Number)*new_alloc_size);
+          values.reset (new_val);
```

Comment:
I must admit I do not know enough about the mechanisms behind `new/delete` versus `malloc/free`. We used `free` before on the memory we got from `posix_memalign` but now we go through the `delete` path if am not mistaken because `unique_ptr` cannot know that we allocated with a `malloc` type call rather than `new`. Now `operator new` definitely goes through the `malloc` interface, but I am not sure if `posix_memalign` does something in addition. Do you know @bangerth? On the other hand, how would `free` know something else than going directly to `delete`, so I guess this is fine. Do all MPI tests pass?

Reply:
Yes, this all passed the testsuite. But you're right, I should make sure that we call `free`. Let me work on this for a minute.

Reply:
For reference, the C++11 standard doesn't specify whether `new`/`delete` go through `malloc`/`free` and explicitly mentions that it does not specify this. So the right thing to do is really to make sure we call `free`.